lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -253,34 +278,68 @@ def _disable_if_needed(self, plugin: OpenSearchPlugin) -> bool:
         except KeyError as e:
             raise OpenSearchPluginMissingConfigError(e)
 
+    def _compute_settings(
+        self, config: OpenSearchPluginConfig
+    ) -> Tuple[Dict[str, str], Dict[str, str]]:
+        """Returns the current and the new configuration."""
+        current_settings = ClusterTopology.get_cluster_settings(
```

Comment:
Perhaps this can be retrieved before everything in the `run()` method and passed around in all methods that rely on it (the response payload is very big), we could probably get away with downloading/loading it once and the diffs should still be well computed

Reply:
@Mehdi-Bendriss yes, +1. That was my plan for the /cluster/setting, just like we were discussing to do with the `opensearch-plugin list`.

Reply:
I am moving this task to DPE-3603.